pintc/src/predicate/optimize.rs

```diff
@@ -1,10 +1,12 @@
 mod dead_code_elimination;
 use dead_code_elimination::dead_code_elimination;
 
+use crate::error::Handler;
+
 impl super::Contract {
-    pub fn optimize(mut self) -> Self {
+    pub fn optimize(mut self, handler: &Handler) -> Self {
         // Remove dead code
-        dead_code_elimination(&mut self);
+        let _ = dead_code_elimination(&mut self, handler);
 
         self
     }
```

Comment:
@mohammadfawaz I currently don't have `optimize()` return `WarningEmitted`. I didn't see any use for the result right now. We simply optimize, emit warnings if we have any, and move on. There isn't any conditional compilation from a warning being emitted.

Open to your thoughts on this ofc.

Reply:
This makes sense to me!